CRM/Price/BAO/PriceSet.php

```diff
@@ -737,7 +737,7 @@ public static function processAmount($fields, &$params, &$lineItem, $component =
         case 'Radio':
           //special case if user select -none-
           if ($params["price_{$id}"] <= 0) {
-            continue;
+            continue 2;
```

Comment:
I think this should probably be a break too

Reply:
Ah, cool, I like the thought of extracting the whole thing